src/XR/features/WebXRLayers.ts

```diff
@@ -0,0 +1,222 @@
+import { WebXRFeatureName, WebXRFeaturesManager } from "../webXRFeaturesManager";
+import { WebXRSessionManager } from "../webXRSessionManager";
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
+import { Nullable } from "../../types";
+import { WebXRRenderTargetProvider } from "../webXRRenderTargetProvider";
+import { WebGLHardwareTexture } from "../../Engines/WebGL/webGLHardwareTexture";
+import { RenderTargetTexture } from "../../Materials/Textures/renderTargetTexture";
+import { WebXRCompositionLayerType, WebXRLayerWrapper, WebXRLayerWrapperProvider } from "../webXRLayerWrapper";
+import { Viewport } from "../../Maths/math.viewport";
+
+/**
+ * Provides render target textures and other important rendering information for a given XRCompositionLayer.
+ * @hidden
+ */
+class XRCompositionLayerRenderTargetProvider extends WebXRRenderTargetProvider {
+    protected _lastSubImages = new Map<XREye, XRWebGLSubImage>();
+    private _glContext: WebGLRenderingContext;
```

Comment:
Seems like it's enough, it's only used for the `WebGLHardwareTexture` constructor, I could make it `WebGLRenderingContext | WebGL2RenderingContext` if that feels better?

Reply:
It does match the type of the `._gl` variable, but this is just cosmetic. If you make a change would be great if you add this but very low prio for this one

Reply:
I'm actually going to keep it `WebGLRenderingContext` for now because if I made it `WebGLRenderingContext | WebGL2RenderingContext` I'd have to end up casting it down for the `WebGLHardwareTexture` constructor, which only takes `WebGLRenderingContext`.